src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1378,7 +1380,13 @@ public override T GetFieldValue<T>(int ordinal)
             }
             catch (NpgsqlSafeReadException e)
             {
-                throw e.OriginalException;
+                var writtenBytes = Buffer.ReadPosition - position;
+                var remainingBytes = ColumnLen - writtenBytes;
+                if (remainingBytes > 0)
+                    Buffer.Skip(remainingBytes, false).GetAwaiter().GetResult();
+
+                ExceptionDispatchInfo.Capture(e.OriginalException).Throw();
```

Comment:
You can simply use `Throw(Exception)` method which internally will do `Capture`.

```suggestion
                ExceptionDispatchInfo.Throw(e.OriginalException);
```

Reply:
Yes, I initially did that but unfortunately it is only supported since netstandard2.1 😞